src/libopenrave-core/generictrajectory.cpp

```diff
@@ -484,6 +498,24 @@ class GenericTrajectory : public TrajectoryBase
             /* Read trajectory data */
             ReadBinaryVector(I, this->_vtrajdata);
             ReadBinaryString(I, __description);
+
+            // versions >= 0x0002 have readable interfaces
+            if (versionNumber >= 0x0002)
```

Comment:
@strmio you have to reset any current readable interfaces in the trajectory...

Reply:
All right. By the way behaviours of `deserialize()` for already initialised `Interface` fields are different between the text version and the binary version. While the binary one always resets these fields regardless whether those are given in a trajectory file or not, the text one does not. For example, the text version appends readable interfaces in a file to existing ones without wiping them, and leave the previously set `Description` if not specified in the file. Probably `deserialize()` for the text format also should reset these fields.

Reply:
@strmio i agree, let's fix the text deserialization also